mnelab/mainwindow.py

```diff
@@ -264,15 +264,21 @@ def __init__(self, model):
             self.actions["toolbar"].setChecked(False)
 
         # set up data model for sidebar (list of open files)
-        self.names = QStringListModel()
-        self.names.dataChanged.connect(self._update_names)
-        splitter = QSplitter()
-        self.sidebar = QListView()
+        self.sidebar = QListWidget()
         self.sidebar.setFrameStyle(QFrame.NoFrame)
         self.sidebar.setFocusPolicy(Qt.NoFocus)
-        self.sidebar.setModel(self.names)
+        self.sidebar.setAcceptDrops(True)
+        self.sidebar.setDragEnabled(True)
+        self.sidebar.setDragDropMode(QListWidget.InternalMove)
+        self.sidebar.setDefaultDropAction(Qt.DropAction.MoveAction)
+        self.sidebar.setEditTriggers(QListWidget.DoubleClicked)
+        self.sidebar.model().rowsMoved.connect(self._sidebar_move_event)
```

Comment:
The docs state that this is a private signal, which is not a bad thing per se, but I was wondering if using `beginMoveRows` would be a better idea? Would that also improve the docstring of [`_sidebar_edit_event`](https://doc.qt.io/qt-6/qabstractitemmodel.html#beginMoveRows) (I don't like the two `???` descriptions)?

Reply:
`beginMoveRows` isn't a signal, so if I understand correctly this would require to subclass `QAbstractItemModel`?

ad ???: Qt is still too much magic for me :D Are `parent` and `destination` just references to the source an target model of the drag and drop operation, i.e. the sidebar's underlying model?

Reply:
Hm, `QListWidget` has a const `model` method (which is a `QAbstractItemModel`). I thought that this could be used, but to be honest I don't really know how. I guess if it works the way you implemented it that's fine.

> Are parent and destination just references to the source an target model of the drag and drop operation,

I'd say yes. Not sure if it's necessary to document all parameters, because this function gets called by a signal and not by our code. I also wonder why `flake8` doesn't complain about unused parameters (maybe the docstring should reflect which parameters are unused).

Reply:
> I also wonder why `flake8` doesn't complain about unused parameters

Apparently that's not part of the pure flake8 [error codes](https://flake8.pycqa.org/en/latest/user/error-codes.html), but requires a [plugin](https://github.com/nhoad/flake8-unused-arguments).

> maybe the docstring should reflect which parameters are unused

👍